Documentation/design-docs/arm64-intrinsics.md

```diff
@@ -0,0 +1,527 @@
+  + This is primarily driven by availability of hardware.  Features released in earlier will be prevalent in
+  more hardware.
++ Priorities will be driven by optimization efforts and requests
+  + Priority will be given to intrinsics which are equivalent/similar to those actively used in libraries for other platforms
+  + Priority will be given to intrinsics which have already been implemented for other platforms
+
+### API review
+
+Intrinsics will extend the API of CoreCLR.  They will need to follow standard API review practices.
+
+XArch intrinsics are being added to the NetStandard API https://github.com/dotnet/corefx/issues/24346.  ARM64 intrinsics should be
+treated identically.
+
+#### API review of a new intrinsic `static class`
+
+Review will be facilitated by GitHub Pull requests to amend the Approved API section of this document.
+
+A separate GitHub Issue will typically created for the review of each intrinsic `static class`.  This allows design and review team to
+review separately.  This allows review complexity to be kept manageable.  The O(N^2) nature of the review process will be kept to
+reasonable levels and iterations will be finite.
+
+Every effort will be made to completely elaborate all the methods of the intrinsic `static class` for each review.  This will
+help minimize reopening the same classes for review.
+
+Implementation will be kept separate from from API review.
+
+### Partial implementation of intrinsic `static class`
+
++ `IsSupported` must represent the state of an entire intrinsic `static class`
++ Once API review is complete and approved, it is acceptable to implement approved methods in any order provided tests are added
++ The approved API must be completed before the intrinsic `static class` is included in a release
+
+### Test coverage
+
+As intrinsic support is added test coverage must be extended to provide basic testing
+
+## Half precision floating point
+
+This document will refer to half precision floating point as `Half`.
+
++ Machine learning and Artificial intelligence often use `Half` type to simplify storage and improve processing time.
++ CoreCLR and `CIL` in general do not have general support for a `Half` type
++ There is an open request to expose `Half` intrinsics
++ There is an outstanding proposal to add `System.Half` to support this request https://github.com/dotnet/corefx/issues/25702
++ Implementation of `Half` features will be adjusted based on
+  + Implementation of the `System.Half` proposal
+  + Availability of supporting hardware (extensions)
+  + General language extensions supporting `Half`
+
+### ARMv8 Half precision support
+
+ARMv8 baseline support for `Half` is limited.  The following operations are supported
+
++ Loads and Stores
++ Conversion to/from `Float`
++ Widening from `Vector128<Half>` to two `Vector128<Float>`
++ Narrowing from two `Vector128<Float>` to `Vector128<Half>`
+
+Recent extensions add support for
+
++ General operations on `Half` types
```

Comment:
Perhaps it could be helpful to say that arithmetic in older implementations is not conforming to IEE 754

Reply:
This statement applies to ARM32, but not ARM64 as there is no older implementation

Reply:
I'll add a comment though